### PR TITLE
Introduce Stitches-based Design System 🎨

### PR DESCRIPTION
**Pull request by joe-bell**
## Description

This PR introduces our new [Stitches-based](http://stitches.dev/) Design System, along with its first implementation within the Orchest product.

#### Workflow

> No changes are necessary, just run `pnpm i` when cloning the project

1. Our design system is housed in `lib/design-system`, split across 2 packages:
    - `package` - the _actual_ consumable package
       - _which has been mostly copy/pasted - with improvements - from a private orchest project_
    - `sandbox` - the local development environment, including all consumable components and their variants
       - See [updated docs](https://github.com/orchest/orchest/pull/229/files#diff-a9d64d25e3569d572dc76b01b4a07c69ec86c0a4c07300af2abed42a50212dea)
2. Releases are made in **semver** format to distinguish against the consumable product releases.

#### Usage

See [updated docs](https://github.com/orchest/orchest/pull/229/files#diff-a9d64d25e3569d572dc76b01b4a07c69ec86c0a4c07300af2abed42a50212dea)

### Implementation

> Note: the app will look _slightly_ smaller due to a `17px`(?!) body font size being removed 

1. Our first implementation is added to `ParamTree` component, to `<Alert />` users when parameters haven't been added to a job.
2. Our second implementation will be added to a private package, which needs to consume the latest pagination changes to <Alert />

### Housekeeping

- `pnpm` scripts have been updated to build dependencies of `client-` packages automatically (rather than specifically target the `lib`)

## Checklist

- [x] The documentation reflects the changes.
- [ ] I have manually tested the application to make sure the changes don’t cause any downstream issues, which includes making sure `./orchest status --ext` is not reporting failures when Orchest is running.

- [ ] In case I changed one of the services’ `models.py` I have performed the appropriate database migrations.

### CODE REVIEW

File: services/orchest-webserver/client/src/components/ParamTree.jsx [hunk 1]
```diff
@@ -98,9 +99,19 @@ class ParamTree extends React.Component {
...
+                title="This pipeline doesn't have any parameters defined"
+                description={
+                  <>
+                    <Link href="https://orchest.readthedocs.io/en/stable/user_guide/jobs.html#parametrizing-your-pipeline-and-steps">
+                      Learn more
+                    </Link>{" "}
+                    about paramterizing your pipelines and steps.
+                  </>
+                }
+              />
```
Comment (joe-bell) [inline services/orchest-webserver/client/src/components/ParamTree.jsx:114]: Just highlighting this – which is our first implementation within the product

File: services/orchest-webserver/client/src/components/ParamTree.jsx [hunk 2]
```diff
@@ -98,9 +99,19 @@ class ParamTree extends React.Component {
...
+                title="This pipeline doesn't have any parameters defined"
+                description={
+                  <>
+                    <Link href="https://orchest.readthedocs.io/en/stable/user_guide/jobs.html#parametrizing-your-pipeline-and-steps">
+                      Learn more
+                    </Link>{" "}
+                    about parametrizing your pipelines and steps.
+                  </>
+                }
+              />
```
Comment (joe-bell) [inline services/orchest-webserver/client/src/components/ParamTree.jsx:114]: Just highlighting this – which is our first implementation within the product

File: services/orchest-webserver/client/src/styles/main.scss [hunk 3]
```diff
@@ -52,16 +52,10 @@ $pipelineStepWidth: 190px;
...
-  padding: 0;
-  margin: 0;
 }
 
 p {
-  font-size: $defaultFontSize;
-
   &.padding-top {
     padding-top: $generalPadding;
   }
```
Comment (joe-bell) [inline services/orchest-webserver/client/src/styles/main.scss:61]: These styles have been replaced by `globalStyles()` set in the `<DesignSystemProvider />`

File: services/orchest-webserver/client/src/styles/fonts.scss [hunk 4]
```diff
@@ -1,10 +1,19 @@
...
+    U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF;
+}
+
 /* latin */
 @font-face {
   font-family: "Inter";
   font-style: normal;
   font-weight: 400;
-  src: url("../../fonts/Inter/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1ZL7.woff2")
-    format("woff2");
```
Comment (ricklamers) [inline services/orchest-webserver/client/src/styles/fonts.scss:7]: Much better naming :massage:

**Review by ricklamers — APPROVED** [review]

**Comment by ricklamers** [comment]
You're upleveling our component & design system game. @joe-bell is MVP. Approved!

**Pull request merged**